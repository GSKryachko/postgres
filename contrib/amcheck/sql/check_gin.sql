-- minimal test, basically just verifying that amcheck works with GiST
SELECT setseed(1);
CREATE TABLE "gin_check"("Column1" int[]);
INSERT INTO gin_check select array_agg(round(random()*100)) from generate_series(1, 1000000) as i group by i % 100000;
INSERT INTO gin_check select array_agg(100 + round(random()*100)) from generate_series(1, 100) as i group by i % 100;
CREATE INDEX gin_check_idx on "gin_check" USING GIN("Column1");
SELECT gin_index_parent_check('gin_check_idx');

-- cleanup
DROP TABLE gin_check;
