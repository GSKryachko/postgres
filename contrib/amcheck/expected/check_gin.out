-- minimal test, basically just verifying that amcheck works with GiST
SELECT setseed(1);
 setseed 
---------
 
(1 row)

CREATE TABLE gin_check AS SELECT point(random(),s) c FROM generate_series(1,10000) s;
INSERT INTO gin_check SELECT point(random(),s) c FROM generate_series(1,100000) s;
CREATE INDEX gin_check_idx ON gin_check USING gin(c);
SELECT gin_index_parent_check('gin_check_idx');
 gin_index_parent_check 
-------------------------
 
(1 row)

-- cleanup
DROP TABLE gin_check;
